### ISSUE TITLE
How can I retrieve external results?

### ISSUE THREAD

**Issue by mhowell07** [10up/ElasticPress#3294]
### Describe your question

Hi,

Elasticsearch has the capability to read results from multiple sources. Is this something that is also possible with ElasticPress?

I have an external Zendesk knowledgebase and would like the articles to show up in my ElasticPress search results on my WordPress site.

What would be the best way to approach this?

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by felipeelia** [comment]
Hey @mhowell07,

Although that is possible, as the main purpose of ElasticPress is to serve as a bridge between WP and ES, everything else would require some big development efforts.

There are a couple of ways you could have that done:

1. Import those Zendesk knowledgebase articles into WordPress as a CPT. You could import them daily, for example, and let ElasticPress sync them as it would sync any WP content;
2. Create an Elasticsearch index with those articles (the process would be completely separate from WordPress) and use ElasticPress filters to also search in that index. This can be super tricky though, as this index mapping will need to match exactly what we have for WP posts and you would need to use some extra filters to tell EP that the result item is not something coming from WordPress.

I'd recommend option nr. 1, as that is cleaner and easier to both implement and maintain. I hope that helps!

**Comment by mhowell07** [comment]
Hey @felipeelia ,

Thanks for the response.

I have already started implementing something close to option 2 since I submitted the original question. I was able to get posts from Zendesk into the same index as my WordPress posts and show the results in search. I realize this isn't ideal and EP throws a notice that the index mapping is off due to this change.

My plan is to create a  copy of the WP posts index and wipe the content so I have a correctly mapped index to insert Zendesk posts into. 

Did you have specific filters in mind for telling EP that the result item is not something coming from WordPress?
